### ISSUE TITLE
Logging does not generate any output

### ISSUE THREAD

**Issue by w1005444804** [google/glog#1126]
I've been troubled for many days that unable to print log information in sub thread

**Comment by sergiud** [comment]
Please provide some code that shows the problem.

**Comment by w1005444804** [comment]
@sergiud Hi, My code is roughly as follows。
std::thread(&ImageMaskTrainer::training, this).detach();

ImageMaskTrainer::training:
    std::thread(&ImageMaskDataSet::start_prefetch_traindata, this).detach();
    ...
    void ImageMaskDataSet::start_prefetch_traindata() {
		    LOG(INFO) << "start_prefetch_traindata...";//（not print log）
                    ThreadPool thread_pool(...);
                    //do something , include LOG(INFO) (not print log also)
	    }

**Comment by sergiud** [comment]
I still do not understand the problem. Are you saying the specific thread does not print any log? If so how did you check this?

**Comment by w1005444804** [comment]
yes, there is no information printed on the screen in the sub thread。

**Comment by sergiud** [comment]
Just to be sure: you see log output outside of the thread but not when running one, correct?

**Comment by w1005444804** [comment]
@sergiud ，Even if the program ends, I can't see any logs about sub threads

**Comment by sergiud** [comment]
For some reason you keep ignoring my questions by providing only partial answers. Therefore, I'll assume that what you observe is not thread specific. In this case this is expected as glog does not log to stdout/stderr by default. You need to enable this explicitly. Please refer to the [documentation](https://google.github.io/glog/stable/flags/) how to do this.

**Comment by w1005444804** [comment]
@sergiud Sorry, I have been busy with other things in the past few days. My concern is that logs can be printed on the screen in the main thread, but they cannot be printed in the child threads(std::thread(***).detach();) created in the main thread。

**Comment by sergiud** [comment]
I cannot reproduce this. Here's the test code:

```cpp
#include <thread>

#include <glog/logging.h>

int main()
{
    LOG(INFO) << "hello from main";

    std::thread{[] { LOG(INFO) << "hello from thread"; }}.join();
}
```

Output:
```console
WARNING: Logging before InitGoogleLogging() is written to STDERR
I20240914 11:37:32.752182 124638586318400 glog-thread-no-log.cpp:7] hello from main
I20240914 11:37:32.752288 124638579394240 glog-thread-no-log.cpp:9] hello from thread
```
If you believe the issue is in glog, make sure to provide an exact reproducer.

**Comment by w1005444804** [comment]
@sergiud 
#include <thread>

#include <glog/logging.h>

int main()
{
    LOG(INFO) << "hello from main";

    std::thread{[] { LOG(INFO) << "hello from thread"; }}.detach();
}

Just change join() to detach(), then...

WARNING: Logging before InitGoogleLogging() is written to STDERR
I20240914 23:24:08.007007 35596 main.cpp:7] hello from main

**Comment by sergiud** [comment]
That's not a glog issue as explained. Please refer to concurrency basics in C++.
